src/main/java/appeng/client/gui/implementations/GuiCraftConfirm.java

```diff
@@ -111,16 +128,45 @@ boolean isAutoStart()
 		return ( (ContainerCraftConfirm) this.inventorySlots ).isAutoStart();
 	}
 
+	private void reinitalize()
+	{
+		this.buttonList.clear();
+		this.initGui();
+	}
+
 	@Override
 	public void initGui()
 	{
+		this.maxRows = this.getMaxRows();
+
+		final int magicNumber = ITEMSTACK_TOP_OFFSET + CPU_TOP_OFFSET;
+		final int extraSpace = this.height - magicNumber;
+
+		this.rows = (int) Math.floor( extraSpace / SECTION_HEIGHT );
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this silly call to "Math.floor" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2185)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Cast one of the operands of this division operation to a "double". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2184)